### PR TITLE
Allow modified keycodes in Unicode input

### PR DESCRIPTION
**Pull request by fauxpark**
## Description

Especially useful if you have your OS layout set to something non-US ANSI, where the number row is flipped (such as French AZERTY).

For example, you could add the following to your keymap.c:
```c
uint16_t hex_to_keycode(uint8_t hex) {
    switch (hex) {
        case 0:
            return FR_0;
        case 1 ... 9:
            return FR_1 + (hex - 1);
        case 0xA:
            return FR_A;
        case 0xB:
            return FR_B;
        case 0xC:
            return FR_C;
        case 0xD:
            return FR_D;
        case 0xE:
            return FR_E;
        case 0xF:
            return FR_F;
        default:
            return KC_NO;
    }
}
```

## Types of Changes

- [x] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: quantum/process_keycode/process_unicode_common.c [hunk 1]
```diff
@@ -136,20 +136,10 @@ __attribute__((weak)) void unicode_input_cancel(void) {
     set_mods(unicode_saved_mods);  // Reregister previously set mods
 }
 
-__attribute__((weak)) uint16_t hex_to_keycode(uint8_t hex) {
```
Comment (drashna) [inline quantum/process_keycode/process_unicode_common.c:139]: Did you mean to remove this? 

I think that this still needs to be there.

Comment (fauxpark) [reply quantum/process_keycode/process_unicode_common.c:139]: Yes, it already exists in quantum.c. I tested it and it still works :+1:

Comment (drashna) [reply quantum/process_keycode/process_unicode_common.c:139]: Ah, okay. Just making sure.

**Review by tzarc — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Comment by vansoest** [comment]
Is hex_to_keycode renamed or removed? What is the equivalent or how to remap the hex codes in latest release? 

I just tried in my keymap to remap it, but I cannot find any hex_to_keycode function in the recent code.

**Comment by fauxpark** [comment]
https://github.com/qmk/qmk_firmware/pull/12079#issuecomment-788823407

**Pull request merged**